hw/syn/tools/dc/run-syn.tcl

```diff
@@ -0,0 +1,120 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Simple tcl script for DC to do some wire-load-model-based test syntheses.
+
+#####################
+##  PREPARE FLOW   ##
+#####################
+
+# tool setup
+source [getenv "LIB_SETUP_SCRIPT"]
+
+# paths
+set WORKLIB  "WORK"
+set REPDIR   "REPORTS"
+set DDCDIR   "DDC"
+set VLOGDIR  "NETLISTS"
+
+exec mkdir -p ${REPDIR} ${DDCDIR} ${VLOGDIR} ${WORKLIB}
+
+# define work lib path
+define_design_lib WORK -path $WORKLIB
+
+#######################
+##  DESIGN SOURCES  ###
+#######################
+
+# just compile the "core" toplevel at the moment
+# might want to switch to top_earlgrey_asic later on (with pads)
+set TOP_ENTITY [getenv "TOP_ENTITY"]
+
+# read source list generated by fusesoc
+set fp [open [ls "*.scr"] r]
+set file_data [read $fp]
+close $fp
+set data [split $file_data "\n"]
+
+# this TECH_LIB_SEL selects the appropriate technology by defining
+# PRIM_DEFAULT_IMPL=prim_pkg::Impl<tech identifier>
+# TECH_LIB_SEL is set inside the library setup script
+set DEFINE "PRIM_DEFAULT_IMPL=${PRIM_DEFAULT_IMPL} "
+# go through fusesoc file list and separate +defines from actual source files
+set SRC {}
+foreach item $data {
+  if {[string range $item 0 7] == "+define+"}  {
+    set DEFINE "${DEFINE}[string range $item 8 end] "
+  } elseif {[string range $item 0 7] == "+incdir+"}  {
+    lappend search_path "[string range $item 8 end]"
+  } else {
+    lappend SRC $item
+  }
+}
+
+# additional parameters
+set PARAMS ""
+
+###########################
+##   ELABORATE DESIGN    ##
+###########################
+
+# delete previous designs.
+remove_design -designs
+sh rm -rf $WORKLIB/*
+
+analyze -define ${DEFINE} -format sv ${SRC}
+elaborate  ${TOP_ENTITY} -parameters ${PARAMS}
+link > "${REPDIR}/${TOP_ENTITY}_link.rpt"
+
+write_file -format ddc -hierarchy -output "${DDCDIR}/${TOP_ENTITY}_elab.ddc"
```

Comment:
if it's still allowed, you can write_file -format verilog here if you need to debug elaboration, ever.

Reply:
yeah good point, let me add that.

Reply:
done